api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -198,16 +205,19 @@ export class ProjectAclService implements ProjectAccessControl {
     await apiQueryRunner.startTransaction();
 
     try {
-      const existingUserInProject = await apiQueryRunner.manager.findOne(
-        UsersProjectsApiEntity,
-        undefined,
-        {
-          where: {
-            projectId,
-            userId,
-          },
-        },
-      );
+      const existingUserInProject = await apiQueryRunner.manager
+        .createQueryBuilder(UsersProjectsApiEntity, 'users_projects')
+        .where({
+          projectId,
+          userId,
+        })
+        .leftJoinAndSelect('users_projects.user', 'userId')
+        .select(['users_projects.roleName', 'userId.isDeleted'])
+        .getOne();
```

Comment:
another way to address this may be to make the "user has a role but user is deleted" case as an "anomaly" and only expose to the rest of the logic whether an active user already has a role, e.g.

```suggestion
      const isUserActiveAndHasAlreadyARoleInProject = await apiQueryRunner.manager
        .createQueryBuilder(UsersProjectsApiEntity, 'users_projects')
        .where({
          projectId,
          userId,
        })
        .leftJoinAndSelect('users_projects.user', 'userId')
        .select(['users_projects.roleName', 'userId.isDeleted'])
        .getOne()
        .then(user => {
          // handle the user is deleted case here
          // otherwise return true if the user is active and has a role, false if it doesn't have a role
        })
        .catch(...);
```

Reply:
btw don't feel like you have to go down this madness route 😁

Reply:
By keeping `.select(['users_projects.roleName', 'userId.isDeleted']).getOne()` and then adding the `then()`would not it be already filtering out the deleted user?

I assume there is a mistake in your suggestion or I don't totally understand what you meant here.

Reply:
in the `.then()`, the parameter received should be the found `UsersProjectsApiEntity` (if found, and only the selected fields) - why would the result we get in the `.then()` be different than what we currently store in `existingUserInProject`?

but yes, let's not go down this madness route 😅